app/src/main/java/io/homeassistant/companion/android/sensors/SensorsSettingsFragment.kt

```diff
@@ -15,6 +16,31 @@ class SensorsSettingsFragment : PreferenceFragmentCompat() {
     @Inject
     lateinit var integrationUseCase: IntegrationUseCase
 
+    private val handler = Handler()
+    private val refresh = object : Runnable {
+        override fun run() {
+            val sensorDao = AppDatabase.getInstance(requireContext()).sensorDao()
+            SensorReceiver.MANAGERS.plus(LocationBroadcastReceiver()).forEach { managers ->
+                managers.requestSensorUpdate(requireContext())
+                managers.availableSensors.forEach { basicSensor ->
+                    findPreference<Preference>(basicSensor.id)?.let {
+                        val sensorEntity = sensorDao.get(basicSensor.id)
+                        if (sensorEntity?.enabled == true) {
+                            if (basicSensor.unitOfMeasurement.isNullOrBlank())
+                                it.summary = sensorEntity.state
+                            else
+                                it.summary = sensorEntity.state + " " + basicSensor.unitOfMeasurement
+                            // TODO: Add the icon from mdi:icon?
+                        } else {
+                            it.summary = "Disabled"
+                        }
+                    }
+                }
+            }
+            handler.postDelayed(this, 10000)
```

Comment:
Does this mean if we keep the settings screen open we can get updates every 10 seconds into HA?

Reply:
That is correct!  We can consider doing something similar to the iOS app add add this to the webview activity as well to get more consistent updates when the application is open.

Reply:
That would be amazing for wall mounted tablets!!!